src/main/java/seedu/pill/util/Parser.java

```diff
@@ -117,6 +150,202 @@ public void parseCommand(String input) {
         }
     }
 
+    /**
+     * Parses the arguments provided to create a {@link FulfillCommand} instance, which is used to fulfill an order.
+     *
+     * @param arguments The command input containing the order index to be fulfilled.
+     * @return A {@link FulfillCommand} instance that contains the order and transaction manager.
+     * @throws PillException if the input contains too many arguments, is empty, cannot be parsed as a number,
+     *                       or if the specified order index is invalid.
+     *
+     *                       <p>Specifically, this method throws:</p>
+     *                       <ul>
+     *                           <li>{@code PillException} with {@code ExceptionMessages.TOO_MANY_ARGUMENTS} if more
+     *                           than one argument is provided.</li>
+     *                           <li>{@code PillException} with {@code ExceptionMessages.INVALID_FULFILL_COMMAND} if no
+     *                           argument is provided.</li>
+     *                           <li>{@code PillException} with {@code ExceptionMessages.INVALID_INDEX} if the argument
+     *                           cannot be parsed as an integer.</li>
+     *                           <li>{@code PillException} with {@code ExceptionMessages.INVALID_ORDER} if the specified
+     *                           order index is out of bounds.</li>
+     *                       </ul>
```

Comment:
The JavaDocs here seems a bit odd here. Perhaps we can shorten this, and move the existing details in full to the DG.

Reply:
Edited in [79e3c2b](https://github.com/AY2425S1-CS2113-W14-4/tp/pull/142/commits/79e3c2bb8551cf52774c68cddb0a1e055f37a211)